test/acceptance/uk/gov/hmrc/apidocumentation/BaseSpec.scala

```diff
@@ -18,38 +18,42 @@ package acceptance.uk.gov.hmrc.apidocumentation
 
 import com.github.tomakehurst.wiremock.WireMockServer
 import com.github.tomakehurst.wiremock.client.WireMock
-import com.github.tomakehurst.wiremock.core.WireMockConfiguration
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig
 import org.openqa.selenium.WebDriver
 import org.scalatest._
-import org.scalatestplus.play.OneServerPerSuite
+import org.scalatestplus.play.guice.{GuiceFakeApplicationFactory, GuiceOneServerPerSuite}
 import play.api.inject.guice.GuiceApplicationBuilder
 import play.api.{Application, Mode}
 
-trait TestSpec extends FeatureSpec with BeforeAndAfterEach with BeforeAndAfterAll with Matchers with NavigationSugar {
-  implicit val webDriver: WebDriver = Env.driver
+trait BaseSpec extends FeatureSpec with BeforeAndAfterEach with BeforeAndAfterAll with Matchers with NavigationSugar
+  with GuiceOneServerPerSuite with GuiceFakeApplicationFactory {
 
+  override lazy val port = Env.port
+  val stubPort = 11111
   val stubHost = "localhost"
-  val stubPort = sys.env.getOrElse("WIREMOCK_PORT", "11111").toInt
-  val wireMockServer = new WireMockServer(WireMockConfiguration.wireMockConfig().port(stubPort))
 
-  override def beforeAll() = {
-    if (!wireMockServer.isRunning) {
-      wireMockServer.start()
-    }
+  implicit val webDriver: WebDriver = Env.driver
+
+  var wireMockServer = new WireMockServer(wireMockConfig().port(stubPort))
+
+  override def fakeApplication(): Application = {
+    GuiceApplicationBuilder()
+      .configure("run.mode" -> "Stub")
```

Comment:
I get an IntelliJ red underline on this, but don't if it is changed to longhand as an explicitly typed Map or to a (x,y) tuple.  It obviously compiles in sbt but it would be good not to have the red line (perhaps it's my environment).

Reply:
There is a predef implicit that should handle this for you :(
https://docs.scala-lang.org/overviews/collections/maps.html